File: govardata/datavar.go
package main

import (
	"fmt"
)

func main() {
	// Integer
	var myInt int = 42
	fmt.Printf("Integer: %d\n", myInt)

	// Long (int64)
	var myLong int64 = 123456789
	fmt.Printf("Long: %d\n", myLong)

	// Float
	var myFloat float32 = 3.14
	fmt.Printf("Float: %f\n", myFloat)

	// Double
	var myDouble float64 = 3.14159
	fmt.Printf("Double: %f\n", myDouble)

	// String
	var myString string = "Hello, Go!"
	fmt.Printf("String: %s\n", myString)

	// Boolean
	var myBoolean bool = true
	fmt.Printf("Boolean: %t\n", myBoolean)

	// Char (rune)
	var myChar rune = 'A'
	fmt.Printf("Char: %c\n", myChar)

	// Byte
	var myByte byte = 127
	fmt.Printf("Byte: %d\n", myByte)

	// Short (int16)
	var myShort int16 = 32767
	fmt.Printf("Short: %d\n", myShort)

	// Array
	var myArray = [5]int{1, 2, 3, 4, 5}
	fmt.Printf("Array: %v\n", myArray)

	// Slice
	var mySlice = []int{1, 2, 3, 4, 5}
	fmt.Printf("Slice: %v\n", mySlice)

	// Map
	var myMap = map[string]int{"one": 1, "two": 2, "three": 3}
	fmt.Printf("Map: %v\n", myMap)

	// Function
	myFunction := func(x int) int {
		return x * 2
	}
	fmt.Printf("Function (2 * 3): %d\n", myFunction(3))

	// Struct
	type Person struct {
		Name string
		Age  int
	}
	var myPerson = Person{"Alice", 25}
	fmt.Printf("Struct: %v\n", myPerson)

	// Nil
	var myNilPointer *int = nil
	fmt.Printf("Nil: %v\n", myNilPointer)
}

File: datavar.rs
fn main() {
    // Integer
    let my_int: i32 = 42;
    println!("Integer: {}", my_int);

    // Long (i64)
    let my_long: i64 = 123456789;
    println!("Long: {}", my_long);

    // Float
    let my_float: f32 = 3.14;
    println!("Float: {}", my_float);

    // Double
    let my_double: f64 = 3.14159;
    println!("Double: {}", my_double);

    // String
    let my_string: &str = "Hello, Rust!";
    println!("String: {}", my_string);

    // Boolean
    let my_boolean: bool = true;
    println!("Boolean: {}", my_boolean);

    // Char
    let my_char: char = 'A';
    println!("Char: {}", my_char);

    // Byte
    let my_byte: u8 = 127;
    println!("Byte: {}", my_byte);

    // Short (i16)
    let my_short: i16 = 32767;
    println!("Short: {}", my_short);

    // Array
    let my_array: [i32; 5] = [1, 2, 3, 4, 5];
    println!("Array: {:?}", my_array);

    // Vector
    let my_vector: Vec<i32> = vec![1, 2, 3, 4, 5];
    println!("Vector: {:?}", my_vector);

    // HashMap
    use std::collections::HashMap;
    let mut my_map = HashMap::new();
    my_map.insert("one", 1);
    my_map.insert("two", 2);
    my_map.insert("three", 3);
    println!("HashMap: {:?}", my_map);

    // Tuple
    let my_tuple: (i32, &str, bool) = (42, "Rust", true);
    println!("Tuple: {:?}", my_tuple);

    // Option
    let my_option: Option<i32> = Some(42);
    let my_none_option: Option<i32> = None;
    println!("Option (Some): {:?}", my_option);
    println!("Option (None): {:?}", my_none_option);

    // Function
    let my_function = |x: i32| -> i32 { x * 2 };
    println!("Function (2 * 3): {}", my_function(3));

    // Struct
    struct Person {
        name: String,
        age: i32,
    }
    let my_person = Person {
        name: String::from("Alice"),
        age: 25,
    };
    println!("Struct: {:?}", my_person);
}
